src/Lean/Meta/Tactic/NormCast.lean

```diff
@@ -0,0 +1,162 @@
+/-
+Copyright (c) 2019 Paul-Nicolas Madelaine. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Paul-Nicolas Madelaine, Robert Y. Lewis, Mario Carneiro, Gabriel Ebner
+-/
+import Lean.Meta.CongrTheorems
+import Lean.Meta.Tactic.Simp.SimpTheorems
+import Lean.Meta.CoeAttr
+
+namespace Lean.Meta.NormCast
+/--
+`Label` is a type used to classify `norm_cast` lemmas.
+* elim lemma:   LHS has 0 head coes and ≥ 1 internal coe
+* move lemma:   LHS has 1 head coe and 0 internal coes,    RHS has 0 head coes and ≥ 1 internal coes
+* squash lemma: LHS has ≥ 1 head coes and 0 internal coes, RHS has fewer head coes
+-/
+inductive Label
+  /-- elim lemma: LHS has 0 head coes and ≥ 1 internal coe -/
+  | elim
+  /-- move lemma: LHS has 1 head coe and 0 internal coes,
+  RHS has 0 head coes and ≥ 1 internal coes -/
+  | move
+  /-- squash lemma: LHS has ≥ 1 head coes and 0 internal coes, RHS has fewer head coes -/
+  | squash
+  deriving DecidableEq, Repr, Inhabited
```

Comment:
Can we arrange there to be a single source of truth for this info? E.g. with careful use of `inherit_doc`?